### DISCUSSION TITLE
Support for "header id"? {#section-architecture-constraints}

### DISCUSSION THREAD

**Discussion by andife** [executablebooks/meta#816] [Q&A] [answered]
Hello, 
I want to use the markdown files from https://arc42.org/download#file-based-formats for documentation of architecture.

The file contains the following entries.

```
# Architecture Constraints {#section-architecture-constraints}

# System Scope and Context {#section-system-scope-and-context}

## Business Context {#_business_context}
```

Did someome was able to correctly process {#_business_context}. In my case it is written as normal text. Reading the myst documentation, I do not see a support for this? What kind of markdown style is this? A standard one? For me it seems like to be a header id? (https://uhded.com/add-id-header-markdown)

**Comment by chrisjsewell** [comment, marked as answer]
Heya, the myst syntax for this is:

```markdown
(section-architecture-constraints)=
# Architecture Constraints

(section-system-scope-and-context)=
# System Scope and Context

(_business_context)=
## Business Context
```

**Reply by andife** [reply]
Thank you, I will probably change the code accordingly and try if the links work. 

What kind of support would I have to wait for in order to have appropriate standard support?
Is there already an issue? (if it does not go through a configuration? for this issue, or is that too specific?

**Reply by chrisjsewell** [reply]
> What kind of support would I have to wait for in order to have appropriate standard support?

Well, when you say "standard support" this is not a standard Markdown syntax (i.e. in https://spec.commonmark.org/) 😅 

There is some thinking, by me, to change them to something similar to what you show, BUT as per how they are at present, they would still be above the heading, e.g.

```markdown
{#section-architecture-constraints}
# Architecture Constraints
```

This is inline with recent thinking on a "Markdown v2" (see https://johnmacfarlane.net/beyond-markdown.html#attributes)

**Comment by andife** [comment]
Thank you for the detailed answer and also the links. 

I didn't have the "complicatedness" on the screen and didn't know 
the technical terms I would have had to search for.
